pom.xml

```diff
@@ -55,6 +55,16 @@ limitations under the License.
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-api-xml</artifactId>
+      <version>4.0.0-alpha-4</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-xml-impl</artifactId>
```

Comment:
this one https://github.com/apache/maven/blob/maven-4.0.0-alpha-4/maven-xml-impl/pom.xml depends on sisu plexus
and sisu plexus depends on many many things, including plexus-utils (not provided, as at least maven-xml-impl did as a workaround for these circular dependencies): https://github.com/eclipse/sisu.plexus/blob/releases/0.3.5/org.eclipse.sisu.plexus/pom.xml

this dependency adds a dependency hell IMHO

why does maven-xml-impl require sisu plexus?

Reply:
Because of [this class](https://github.com/apache/maven/blob/a00a44f75b8734202514ab5f911b389b80557b2e/maven-xml-impl/src/main/java/org/apache/maven/internal/xml/XmlPlexusConfiguration.java) which depends on sisu.  It could be moved out of `maven-xml-impl` I suppose.

Reply:
The class is [used](https://github.com/apache/maven/blob/a00a44f75b8734202514ab5f911b389b80557b2e/maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/PluginDescriptorBuilder.java) in `maven-plugin-api` unfortunately, so it can't be moved easily into `maven-core`.